### PR TITLE
add stub user account confirmation page and tests for error cases

### PR DESCRIPTION
**Pull request by czue**
Related to [ongoing covid signup work](https://docs.google.com/document/d/1gZlOGroRTeUMHXwzCuOVw2EUuJcjOsAaAe1qc2iUEe0/edit)

Didn't get very far on this today due to other distractions, though it's the first time I'm using TDD on view code and overall I'm enjoying the experience... 

(Going to tag these as invisible until the feature is done to avoid confusing the product team)

### CODE REVIEW

File: corehq/apps/users/views/mobile/users.py [hunk 1]
```diff
@@ -1396,3 +1398,33 @@ def post(self, request, *args, **kwargs):
             self.invitation.registered_date = datetime.utcnow()
             self.invitation.save()
         return self.get(request, *args, **kwargs)
+
+
+@method_decorator(TWO_STAGE_USER_PROVISIONING.required_decorator(), name='dispatch')
+class CommCareUserConfirmAccountView(TemplateView, DomainViewMixin):
+    template_name = "users/commcare_user_confirm_account.html"
+    urlname = "commcare_user_confirm_account"
+    strict_domain_fetching = True
```
Comment (snopoke) [inline corehq/apps/users/views/mobile/users.py]: why is this necessary?

**Review by snopoke — APPROVED** [review]

**Review by esoergel — APPROVED** [review]

File: corehq/apps/users/templates/users/commcare_user_confirm_account.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  {% if user.is_account_confirmed and user.is_active %}
+    <div>
+      <p>
+        {% blocktrans %}
+          Your account is already confirmed! Go sign in.
+        {% endblocktrans %}
+      </p>
+    </div>
+  {% elif user.is_account_confirmed %}
+    <div class="alert-alert-danger">
```
Comment (orangejenny) [inline corehq/apps/users/templates/users/commcare_user_confirm_account.html]: `alert-alert-danger` => `alert alert-danger`

File: corehq/apps/users/templates/users/commcare_user_confirm_account.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      </p>
+    </div>
+  {% else %}
+    <div id="confirm-account-form">
+      <p>Confirm your account here!</p>
+      {#  todo    {% crispy form %}#}
+    </div>
+  {% endif %}
+{% endblock %}
+{% block mobile-logout %}{% endblock %}
```
Comment (orangejenny) [inline corehq/apps/users/templates/users/commcare_user_confirm_account.html]: What's this for? Was this template previously extending `hqwebapp/base_mobile.html`?

File: corehq/apps/users/templates/users/commcare_user_confirm_account.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <h1>{% blocktrans %}{{ hr_name }} Registration{% endblocktrans %}</h1>
+  </div>
+{% endblock %}
+
+{% block content %}
+  {% if user.is_account_confirmed and user.is_active %}
+    <div>
+      <p>
+        {% blocktrans %}
+          Your account is already confirmed! Go sign in.
```
Comment (orangejenny) [inline corehq/apps/users/templates/users/commcare_user_confirm_account.html]: What if the view just redirected already-confirmed users straight to login, with a message like this?

File: corehq/apps/users/templates/users/commcare_user_confirm_account.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      <p>
+        {% blocktrans %}
+          It looks like your account has been deactivated.
+          Contact your administrator if you think this is a mistake.
+        {% endblocktrans %}
+      </p>
+    </div>
+  {% else %}
+    <div id="confirm-account-form">
+      <p>Confirm your account here!</p>
```
Comment (orangejenny) [inline corehq/apps/users/templates/users/commcare_user_confirm_account.html]: translate

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
screenshot?

**Review by millerdev — APPROVED** [review]

File: corehq/apps/users/templates/users/commcare_user_confirm_account.html (hunk 3 above)
Comment (czue) [reply corehq/apps/users/templates/users/commcare_user_confirm_account.html]: oops, yeah I coped from `corehq/apps/users/templates/users/mobile/commcare_user_self_register.html`. will remove

File: corehq/apps/users/views/mobile/users.py (hunk 1 above)
Comment (czue) [reply corehq/apps/users/views/mobile/users.py]: Initially I copied it from the view I was working from, then figured out what it did, and then decided to keep it because:

1. if you change any password / settings on the domain (e.g. https://github.com/dimagi/commcare-hq/pull/26864#discussion_r394818458) I want to make sure those reflect immediately
2. I wouldn't expect this view to be accessed regularly so it seems fine to bust the cache

(I assume the only reason this flag exists is because we don't trust our domain cachebusting)

**Comment by czue** [comment]
@orangejenny I'll follow up on your feedback and post a screenshot in the next PR when I actually have a UI. merging now to make future reviews easier.
